test/parallel/test-diagnostics-channel-module-esm-load.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const dc = require('diagnostics_channel');
+
+let seen;
+dc.subscribe('module.esm.load', common.mustCall((module) => {
+  seen = module;
+}));
+
+import('http').then(http => {
```

Comment:
Can you write a test in ESM (in an .mjs file) to confirm that works too? What about static `import` statements?

Reply:
Not sure how that would work. Wouldn't the module graph already be resolved/loaded by the time any code runs to actually register a subscriber? Maybe it could work in a loader?

Reply:
That's why I asked 😄 

We should add a test that shows expected esm behavior.